NCS.DSS.AzureSearchUtility/CreateIndexer/CreateAddressIndexer.cs

```diff
@@ -0,0 +1,135 @@
+                {
+                    await azureSearchService.DataSources.DeleteAsync(searchConfig.AddressSearchConfig.SearchDataSourceName);
+                }
+            }
+            catch (CloudException e)
+            {
+                Console.WriteLine(e.ToString());
+                throw;
+            }
+
+            Console.WriteLine("{0}", "Creating Address Data Source object...\n");
+            var dataSource = DataSourceHelper.CreateDataSource(searchConfig.AddressSearchConfig.SearchDataSourceQuery,
+                searchConfig.AddressSearchConfig.CollectionId,
+                searchConfig.SearchIndexName,
+                searchConfig.AddressSearchConfig.SearchDataSourceName,
+                searchConfig.AddressSearchConfig.ConnectionString);
+
+            try
+            {
+                Console.WriteLine("{0}", "Attempting to Create/Update Address Data Source...\n");
+                await azureSearchService.DataSources.CreateOrUpdateWithHttpMessagesAsync(dataSource);
+            }
+            catch (CloudException e)
+            {
+                Console.WriteLine(e.ToString());
+                throw;
+            }
+
+            Indexer indexer;
+
+            try
+            {
+                indexer = IndexerHelper.CreateIndexer(azureSearchService,
+                    customerSearchIndex,
+                    searchConfig.AddressSearchConfig.SearchIndexerName,
+                    searchConfig.AddressSearchConfig.SearchDataSourceName,
+                    new List<FieldMapping> { new FieldMapping("CustomerId", "CustomerId") });
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine("{0}{1}", "Unable to Create Address Indexer...\n", e);
+                throw;
+            }
+
+            if (indexer == null)
+            {
+                Console.WriteLine("{0}", "Unable to find Indexer...\n");
+                return new HttpResponseMessage(HttpStatusCode.BadRequest);
+            }
+
+            Console.WriteLine("{0}", "Run Address Indexer...\n");
+            try
+            {
+                azureSearchService.Indexers.Run(indexer.Name);
+            }
+            catch (CloudException e)
+            {
+                Console.WriteLine("{0} {1}", "Unable to get data for Address Indexer...\n", e);
+                throw;
+            }
+
+            var running = true;
+            Console.WriteLine("{0}", "Synchronization running...\n");
+            while (running)
+            {
+                IndexerExecutionInfo status = null;
+
+                try
+                {
+                    status = azureSearchService.Indexers.GetStatus(indexer.Name);
+                }
+                catch (Exception ex)
+                {
+                    Console.WriteLine("Error polling for indexer status: {0}", ex.Message);
+                    throw;
+                }
+
+                var lastResult = status.LastResult;
```

Comment:
Might be worth considering if this variable is required or if status.LastResult could used instead. lastRestul doesn't appear to give any more clarity over status.LastResult.